src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -945,18 +952,20 @@ public Solution RemoveAdditionalDocuments(ImmutableArray<DocumentId> documentIds
         /// </summary>
         public Solution RemoveAnalyzerConfigDocument(DocumentId documentId)
         {
-            if (documentId == null)
-            {
-                throw new ArgumentNullException(nameof(documentId));
-            }
-
-            return RemoveAnalyzerConfigDocuments(ImmutableArray.Create(documentId));
+            CheckContainsAnalyzerConfigDocument(documentId);
+            return RemoveAnalyzerConfigDocumentsImpl(ImmutableArray.Create(documentId));
```

Comment:
Why the Impl method in this case? Given RemoveAnalyzerConfigDocuments still has to guard against nulls or invalid ones, it keeps the code simpler?

Reply:
The argument name is different `documentId` vs `documentIds`.